src/profile-logic/import/chrome.js

```diff
@@ -678,10 +680,28 @@ async function processTracingEvents(
       }
 
       if (profileEvent.name === 'CpuProfile') {
-        profile.meta.profilingStartTime =
-          profileEvent.args.data.cpuProfile.startTime / 1000;
+        // Get profile start time from profile URL if available. Timestamp format is what
+        // is generated by the Chrome DevTools UI when saving profiles. Technically, that
+        // timestamp is of profile save instead of profile start, but one can generate a
+        // more accurate start timestamp through e.g. handling the Chrome DevTools
+        // Protocol `Profiler.consoleProfileStarted` event.
+        const timeStampRe =
+          /(\d{4})(\d{2})(\d{2})T(\d{2})(\d{2})(\d{2}).*\.cpuprofile(?:$|\/|\?|#)/;
```

Comment:
> In a regexp `\b` means "a word boundary", which is exactly what you're looking for here.

I beg to differ, `(?:$|\/|\?|#)` implements the concept of "end of URL subdirectory" which is more strict than "end of word". In particular, `\b` treats characters like `%` and `[` as a boundary which could result in profiles being accepted that do not have exactly the `.cpuprofile` extension (for whatever deliberate reason).

That said, practically there probably isn't much point to be strict about this, and perhaps some people do put additional information on a filename after the file extension. I do prefer that a regex be as strict as possible though.

On the other hand, afaik Firefox Profiler doesn't specifically check for the `.cpuprofile` extension, instead it looks at the profile's contents to determine profile type. So for consistency, the regex should just match a timestamp i.e. it should just be `/(\d{4})(\d{2})(\d{2})T(\d{2})(\d{2})(\d{2})/`. What do you think?

> Why do you have this `.*` between the timestamp and `.cpuprofile`?

The basic filename produced (e.g. `CPU-20230928T151126.cpuprofile`) is lacking in what the profile is all about, and the current regex already ignores everything before the timestamp. The `.*` allows the regex to ignore everything after the timestamp before the extension. This is to cater for people who prefer to add identifying information after the timestamp rather than before it.

Reply:
3c625130e4c5168ac02a95250cf522bca6ed9ce2 -- now I think it's best to be consistent with the rest of Firefox Profiler that doesn't specially treat the `.cpuprofile` extension.

Reply:
251b59cd394245f77a36811377edeed198bac8a9 matches on the last timestamp found, just in case someone sticks a timestamp on a folder. I wanted to use [`String.matchAll()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/matchAll), but Flow 0.96.0 doesn't have a definition for it (!).